charts/memgraph-high-availability/templates/data.yaml

```diff
@@ -9,28 +9,85 @@ spec:
   selector:
     matchLabels:
       app: memgraph-data-{{ $data.id }}
+      role: data
   template:
     metadata:
       labels:
         app: memgraph-data-{{ $data.id }}
+        role: data
         instance-type: data
+
     spec:
-    {{ if $.Values.memgraph.affinity.enabled }}
+      {{- if $.Values.memgraph.affinity.enabled }}
       affinity:
+        {{- if $.Values.memgraph.affinity.nodeSelection }}
+        # Node Selection Affinity: Scheduled on nodes with specific label key and value
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: {{ $.Values.memgraph.affinity.roleLabelKey }}
+                operator: In
+                values:
+                - {{ $.Values.memgraph.affinity.dataNodeLabelValue }}
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: role
+                operator: In
+                values:
+                - data
```

Comment:
Shouldn't you use here also - {{ $.Values.memgraph.affinity.dataNodeLabelValue }}

Reply:
Same at few other places

Reply:
no, because data refers to pod